internal/channelz/funcs.go

```diff
@@ -107,6 +112,23 @@ func NewChannelzStorage() {
 		subChannels:      make(map[int64]*subChannel),
 	})
 	idGen.reset()
+	return func() {
```

Comment:
It feel weird to have the cleanup function **returned** when it's not a closure.
channelz is an internal package. It should be OK to add an exported function which blocks until storage is empty.

Also, the function needs a timeout (a const should be good enough).